code/modules/projectiles/projectile/special/meteor.dm

```diff
@@ -1,3 +1,120 @@
+#define DEFAULT_METEOR_LIFETIME 1800
+#define MAP_EDGE_PAD 5
+
+
+GLOBAL_VAR_INIT(meteor_wave_delay, 625) //minimum wait between waves in tenths of seconds
+//set to at least 100 unless you want evarr ruining every round
+
+//Meteors probability of spawning during a given wave
+GLOBAL_LIST_INIT(meteors_normal, list(/obj/projectile/meteor/dust=3, /obj/projectile/meteor/medium=8, /obj/projectile/meteor/big=3, \
+						  /obj/projectile/meteor/flaming=1, /obj/projectile/meteor/irradiated=3)) //for normal meteor event
+
+GLOBAL_LIST_INIT(meteors_threatening, list(/obj/projectile/meteor/medium=4, /obj/projectile/meteor/big=8, \
+						  /obj/projectile/meteor/flaming=3, /obj/projectile/meteor/irradiated=3)) //for threatening meteor event
+
+GLOBAL_LIST_INIT(meteors_catastrophic, list(/obj/projectile/meteor/medium=5, /obj/projectile/meteor/big=75, \
+						  /obj/projectile/meteor/flaming=10, /obj/projectile/meteor/irradiated=10, /obj/projectile/meteor/tunguska = 1)) //for catastrophic meteor event
+
+GLOBAL_LIST_INIT(meteorsB, list(/obj/projectile/meteor/meaty=5, /obj/projectile/meteor/meaty/xeno=1)) //for meaty ore event
+
+GLOBAL_LIST_INIT(meteorsC, list(/obj/projectile/meteor/dust)) //for space dust event
+
+GLOBAL_LIST_INIT(meteors_stress_test, list(/obj/projectile/meteor/tunguska))
+
+GLOBAL_LIST_EMPTY(meteor_circle)
+GLOBAL_LIST_EMPTY(meteor_circle_directions)
+
+
+///////////////////////////////
+//Meteor spawning global procs
+///////////////////////////////
+
+/proc/spawn_meteors(number = 10, list/meteortypes, direction)
+	for(var/i in 1 to number)
+		spawn_meteor(meteortypes, direction)
+
+/proc/spawn_meteor(list/meteortypes, direction)
+	if(!GLOB.meteor_circle || (GLOB.meteor_circle && GLOB.meteor_circle.len == 0))
+		message_admins("No meteor circle exists, generating at radius 100")
+		var/turf/center_of_station = SSmapping.get_station_center()
+		var/circle_radius = 100
+		var/circle_created = FALSE
+		for(var/i in 1 to 5)
+			GLOB.meteor_circle = midpoint_circle_algo(center_of_station.x, center_of_station.y, circle_radius, center_of_station.z)
+			circle_created = TRUE
+			for(var/T in GLOB.meteor_circle)
+				if(!isspaceturf(T))
+					circle_radius += 25
+					message_admins("Circle too small, generating at radius [circle_radius]")
+					circle_created = FALSE
+					break
+				CHECK_TICK
+			if(circle_created)
+				break
+		listclearnulls(GLOB.meteor_circle)
+		var/bad_spawns = 0
+		for(var/T in GLOB.meteor_circle)
+			if(!isspaceturf(T))
+				GLOB.meteor_circle -= T
+				bad_spawns++
+		message_admins("Removed [bad_spawns] bad spawns from the meteor circle. [GLOB.meteor_circle.len] spawn points remain.")
+		GLOB.meteor_circle_directions = list()
+		for(var/dir in GLOB.alldirs)
+			GLOB.meteor_circle_directions[dir2text(dir)] = list()
+		for(var/T in GLOB.meteor_circle)
```

Comment:
https://github.com/tgstation/tgstation/blob/master/.github/CONTRIBUTING.md#use-descriptive-and-obvious-names